File: LeetCode/Java/55-greedy-跳跃游戏.java
/*
 * @lc app=leetcode.cn id=55 lang=java
 *
 * [55] 跳跃游戏
 */

// @lc code=start
class Solution {
    public boolean canJump(int[] nums) {
        int last = nums.length - 1;
        for (int i = nums.length - 1; i >= 0; --i)
            if (i + nums[i] >= last)
                last = i;
        return last == 0;
    }
}
// @lc code=end

File: LeetCode/C/55-greedy-跳跃游戏.cpp
/*
 * @lc app=leetcode.cn id=55 lang=cpp
 *
 * [55] 跳跃游戏
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    bool canJump(vector<int> &nums) {
        int max_pos = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (i > max_pos)
                return false;
            max_pos = max(max_pos, i + nums[i]);
        }
        return true;
    }
};
// @lc code=end
